src/MensaWidget/MWFavoritesTest.class.st

```diff
@@ -0,0 +1,167 @@
+"
+A MWFavoritesTest is a Tests class that test if the liking of food works as expected.
+
+Instance Variables
+	mensaWidget:		<MWFrame>
+
+mensaWidget
+	- a Mensa widget instance on which the testing will be done
+
+"
+Class {
+	#name : #MWFavoritesTest,
+	#superclass : #TestCase,
+	#instVars : [
+		'mensaWidget'
+	],
+	#category : #'MensaWidget-Tests',
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 16:21'
+}
+
+{
+	#category : #helpers,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 15:30'
+}
+MWFavoritesTest >> getAllMeals [
+	
+	| meals |
+	
+	meals := OrderedCollection new.
+	self mensaWidget contentMorph allMorphsDo: [:each | 
+		each color = MWColorTheme primaryColor ifTrue: 
+			[meals add: each]].
+	^ meals
+]
+
+{
+	#category : #helpers,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 15:37'
+}
+MWFavoritesTest >> getNameFor: aMeal [
+	
+	^ (aMeal submorphs at: 1) contents string
+]
+
+{
+	#category : #helpers,
+	#'squeak_changestamp' : 'LC 6/23/2024 19:18'
+}
+MWFavoritesTest >> getStarColorFor: aMeal [
+	
+	^ (self getStarFor: aMeal)  color
+]
+
+{
+	#category : #helpers,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 16:09'
+}
+MWFavoritesTest >> getStarFor: aMeal [
+	
+	^ aMeal submorphs at: 3
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/11/2024 00:19'
+}
+MWFavoritesTest >> mensaWidget [
+
+	^ mensaWidget
+]
+
+{
+	#category : #accessing,
+	#'squeak_changestamp' : 'Louis Christ 6/11/2024 00:19'
+}
+MWFavoritesTest >> mensaWidget: anObject [
+
+	mensaWidget := anObject
+]
+
+{
+	#category : #environment,
+	#'squeak_changestamp' : 'Oscar 6/17/2024 00:14'
+}
+MWFavoritesTest >> setUp [
+
+	self mensaWidget: MWFrame new
+]
+
+{
+	#category : #helpers,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 16:02'
+}
+MWFavoritesTest >> simulateButtonClick: aStar [
+
+	aStar handleMouseDown: 
+		((MouseButtonEvent new setType: #mouseDown position: aStar center buttons: 4 hand: World currentHand) handler: aStar)
+]
+
+{
+	#category : #environment,
+	#'squeak_changestamp' : 'LC 6/23/2024 19:21'
+}
+MWFavoritesTest >> tearDown [
+
+	self mensaWidget owner abandon
+]
+
+{
+	#category : #tests,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 15:26'
+}
+MWFavoritesTest >> testFavouritesIsSet [
+
+	self assert: MWFavorites favouriteFoods class = Set
+]
+
+{
+	#category : #tests,
+	#'squeak_changestamp' : 'Louis Christ 6/22/2024 16:15'
+}
+MWFavoritesTest >> testFavouritesRenderCorrectly [
+	
+	self tearDown;
+		setUp.
```

Comment:
shouldn't this happen automatically?

Reply:
yes but we want it to reload and check that the meals are liked still and rendered based on the liking

Reply:
Yes, but that is exactly what set up and teardown do after each test method. So after the previous method tearDown is called and setUp before the test method

Reply:
You are right I will change this later today

Reply:
or should i do the commit because of cOlLeCtIvE cOdE oWnErShIp

Reply:
still